temba/contacts/models.py

```diff
@@ -1762,22 +1776,24 @@ class ContactURN(models.Model):
                                 help_text="The preferred channel for this URN")
 
     @classmethod
-    def get_or_create(cls, org, contact, urn_as_string, channel=None):
+    def get_or_create(cls, org, contact, urn_as_string, channel=None, extra_path=None):
```

Comment:
The FCM token on mobile device is updated frequently, so, the contact URN should be updated on RapidPRO as well.
The `extra_path` would be used to prevent duplicated contact, updating only path field instead create a new URN or a new contact. This was our first solution. What would be the best way to resolve this problem?